### PR TITLE
Optimize and bugfix

### PR DESCRIPTION
**Pull request by liquidsec**

### CODE REVIEW

File: baddns/lib/httpmanager.py [hunk 1]
```diff
@@ -22,26 +26,43 @@ def __init__(self, target, http_client_class=None, skip_redirects=False):
...
+        for protocol in protocols:
+            base_url = f"{protocol}://{self.target}/"
+            log.debug(f"ready to make request to URL: {base_url}")
+
+            if not self.skip_redirects:
+                tasks.append((self.http_client.get(base_url, follow_redirects=True), base_url))
+
+            tasks.append((self.http_client.get(base_url, follow_redirects=False), base_url))
+
+        tasks_with_urls = [(task, url) for task, url in tasks]
```
Comment (TheTechromancer) [inline baddns/lib/httpmanager.py:41]: tasks_with_urls is identical to tasks?

Comment (liquidsec) [reply baddns/lib/httpmanager.py:41]: tasks is just a list of the task, the tasks_with_urls list is a (task,url) tuple used for deconflicting after they are all done

Comment (TheTechromancer) [reply baddns/lib/httpmanager.py:41]: 🤔

**Review by TheTechromancer — APPROVED** [review]

File: baddns/lib/httpmanager.py (hunk 1 above)
Comment (liquidsec) [reply baddns/lib/httpmanager.py]: i updated this to make it less redundant. I think there was some other reason i did that, and ended up removing leaving the redundant generator left over.

**Pull request merged**